service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ChildSubmissionController.java

```diff
@@ -41,17 +62,45 @@ public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody Callback
             .build();
     }
 
+    @SuppressWarnings("unchecked")
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        if (caseData.getChildren1() != null) {
```

Comment:
do you reckon the controller or the service should handle this logic? I would rather the service handle it I think

Reply:
Yeah I completely agree. I think this was something I intended to clean up but must have forgotten about.